src/core/meshcop/joiner.cpp

```diff
@@ -314,25 +367,29 @@ void Joiner::HandleSecureCoapClientConnect(bool aConnected, void *aContext)
 
 void Joiner::HandleSecureCoapClientConnect(bool aConnected)
 {
-    switch (mState)
+    VerifyOrExit(mState == OT_JOINER_STATE_CONNECT);
+
+    if (aConnected)
     {
-    case OT_JOINER_STATE_CONNECT:
-        if (aConnected)
-        {
-            mState = OT_JOINER_STATE_CONNECTED;
-            SendJoinerFinalize();
-            mTimer.Start(kTimeout);
-        }
-        else
+        SetState(OT_JOINER_STATE_CONNECTED);
+        SendJoinerFinalize();
+        mTimer.Start(kReponseTimeout);
+    }
+    else
+    {
+        JoinerRouter *router = GetNextJoinerRouter();
+
+        while (router != NULL)
         {
-            Complete(OT_ERROR_SECURITY);
+            VerifyOrExit(Connect(*router) != OT_ERROR_NONE);
```

Comment:
Separate this while-loop into a separate method to share with similar code above?

Reply:
:+1: Changed this in the new pushed commit. Added some new code to handle the previous error (from last attempt).